hw03-spring-boot

surdav opened the pull request.

surdav committed:
hw03-spring-boot

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/Hw03SpringBootApplication.java:
```diff
@@ -0,0 +1,23 @@
+	}
+
+	@Bean
+	@Profile("default") // This bean will only be active when the "default" profile is active
```
(опционально) можно удалить указание профиля

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/config/AppConfig.java:
```diff
@@ -0,0 +1,77 @@
+import ru.otus.hw.service.TestRunnerService;
+import ru.otus.hw.service.TestRunnerServiceImpl;
+
+@Configuration
```
Предлагаю перейти на аннотации (Service, Repository, Component)

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/config/AppConfig.java:
```diff
@@ -0,0 +1,77 @@
+public class AppConfig {
+
+    @Bean
+    public AppProperties appProperties(
```
можно удалить
ожидается применение
@ConfigurationProperties(prefix = "test") над AppProperties

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/service/TestRunnerServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+        resultService.showResult(testResult);
+
+        // Output a localized message indicating the test is complete
+        System.out.println(localizationService.getMessage("TestRunner.completed"));
```
осталось заменить  System.out.println на вызов ioService

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/service/TestServiceImpl.java:
```diff
@@ -0,0 +1,66 @@
+    public TestResult executeTestFor(Student student) {
+        ioService.printLine("");
+        // Output localized message for answering questions
+        ioService.printLine(localizationService.getMessage("TestService.answer.the.questions"));
```
Предлагаю добавить LocalizedIOServiceImpl 
и через него выводить
ioService.printLineLocalized

vlad-iv commented on hw03-spring-boot/src/test/java/ru/otus/hw/Hw03SpringBootApplicationTests.java:
```diff
@@ -0,0 +1,17 @@
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest
+@ActiveProfiles("test") // Activates the "test" profile during tests
```
(опционально) можно оставить, Переход на SpringBootTest это следующее задание

vlad-iv commented on hw03-spring-boot/src/test/java/ru/otus/hw/dao/CsvQuestionDaoTest.java:
```diff
@@ -0,0 +1,111 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+class CsvQuestionDaoTest {
```
Предлагаю добавить тест для TestServiceImpl без SpringBootTest
на моках

surdav committed:
hw03-spring-boot correction

surdav committed:
Hw03SpringBootApplicationTests deletion

surdav committed:
refactor checkstyle

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/config/AppConfig.java:
```diff
@@ -0,0 +1,70 @@
+
+@Service
+@ComponentScan({"ru.otus.hw"}) // Scan components in the specified package
+public class AppConfig {
```
Предлагаю перейти с java конфига на аннотациии
1. удалить AppConfig 
2. расставить аннотации над бинами (Service, Component, Repository)

surdav committed:
Switch from Java config to annotations
- Removed AppConfig.java
- Added appropriate annotations (@Service, @Component, @Repository) to beans

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/config/AppProperties.java:
```diff
@@ -0,0 +1,17 @@
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
```
можно удалить

surdav replied:
Спасибо, исправил!

vlad-iv commented on hw03-spring-boot/src/main/java/ru/otus/hw/Hw03SpringBootApplication.java:
```diff
@@ -0,0 +1,21 @@
+import org.springframework.context.annotation.Bean;
+import ru.otus.hw.service.TestRunnerService;
+
+@SpringBootApplication
```
Осталось добавить аннотацию EnableConfigurationProperties

surdav replied:
Спасибо, добавил!

surdav committed:
refactor
